starknet-signers/src/key_pair.rs

```diff
@@ -100,8 +101,8 @@ impl SigningKey {
     }
 
     /// Gets the secret scalar in the signing key.
-    pub const fn secret_scalar(&self) -> Felt {
-        self.secret_scalar
+    pub const fn secret_scalar(&self) -> &Felt {
```

Comment:
What motivated a change to `&` when `Felt` is copiable? Most usage takes a `Felt` as argument which would lead to a copy while dereferencing.

Reply:
For this case, the motivation is to avoid creating another copy of a Felt that will not be zeroized on drop and stay in memory. Returning just a reference to the internal value will solve this problem, the value will still be zeroized at the end.

Reply:
> What motivated a change to `&` when `Felt` is copiable? Most usage takes a `Felt` as argument which would lead to a copy while dereferencing.

For this case, the motivation is to avoid creating another copy of a Felt that will not be zeroized on drop and stay in memory. Returning just a reference to the internal value will solve this problem, the value will still be zeroized at the end.